File: front/FogbankAngular/src/app/models/Candidature.ts
import { AnneeEtude } from "./annee-etude.enum";
import { LangueStatus } from "./langue-status.enum";
import { Offre } from "./offre";
import { OptionEtude } from "./option-etude.enum";
import { StatutCandidature } from "./statut-candidature.enum";
import { User } from "./User";

export interface Candidature {
  id?: number;
  dateDepot: Date; // ISO format: yyyy-mm-dd
  statut: StatutCandidature;

  nom: string;
  prenom: string;
  idEsprit: string;
  phone: string;
  email: string;
  emailesprit: string;
  emailpersonel: string;

  CV: string;
  lettreMotivation: string;
  autreDocs: string;

  anneeActuelle: AnneeEtude;
  optionActuelle: OptionEtude;
  creditsNonValides: boolean;
  niveauLangue: LangueStatus;

  moyenne1A: number;
  moyenne2A: number;
  moyenne3A: number;
  moyenne4A: number;
  moyenne5A: number;

  offre: Offre;
  user: User;

  cvData?: any;  // or use ArrayBuffer if preferred
  lettreMotivationData?: any;
  autreDocsData?: any;

  score?: number; // Add this line
}

File: back/FogbankSpringBootPi/src/main/java/com/fogbank/springsecurity/entities/Candidature.java
package com.fogbank.springsecurity.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@Entity
public class Candidature implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private LocalDate dateDepot;
    @Enumerated(EnumType.STRING)
    private StatutCandidature statut;
    private String nom;
    private String prenom;
    private String idEsprit;
    private String phone;
    private String email;
    private String emailesprit;
    private String emailpersonel;

    private String CV;
    private String lettreMotivation;
    private String autreDocs;

    @Enumerated(EnumType.STRING)
    private AnneeEtude anneeActuelle;
    @Enumerated(EnumType.STRING)
    private OptionEtude optionActuelle;
    private boolean creditsNonValides;
    @Enumerated(EnumType.STRING)
    private LangueStatus niveauLangue;

    private double moyenne1A;
    private double moyenne2A;
    private double moyenne3A;
    private double moyenne4A;
    private double moyenne5A;

    private Double score;

    @Lob
    @Column(columnDefinition = "LONGBLOB")
    private byte[] cvData;

    @Lob
    @Column(columnDefinition = "LONGBLOB")
    private byte[] lettreMotivationData;

    @Lob
    @Column(columnDefinition = "LONGBLOB")
    private byte[] autreDocsData;

    @ManyToOne
    @JsonBackReference(value = "offre-candidature")
    private Offre offre;

    @ManyToOne
    @JsonBackReference(value = "user-candidature")
    private User user;

    @OneToOne(mappedBy = "candidature", cascade = CascadeType.ALL)
    @JsonManagedReference(value = "candidature-administrative-file")
    private AdministrativeFile administrativeFile;

    @OneToOne(mappedBy = "candidature", cascade = CascadeType.ALL)
    @JsonManagedReference(value = "candidature-acceptance-proof")
    private AcceptanceProof acceptanceProof;

    @OneToOne(mappedBy = "candidature", cascade = CascadeType.ALL)
    @JsonIgnoreProperties({"candidature"})
    private MobilityProcess mobilityProcess;

}
